product-catalog/app/com/commercetools/sunrise/productcatalog/productdetail/ProductDetailPageContentFactory.java

```diff
@@ -22,12 +22,15 @@
     protected ProductBeanFactory productBeanFactory;
 
     public ProductDetailPageContent create(final ProductProjection product, final ProductVariant variant) {
-        final ProductDetailPageContent content = new ProductDetailPageContent();
-        fillAdditionalTitle(product, content);
-        fillProduct(product, variant, content);
-        fillBreadCrumb(product, variant, content);
-        fillAddToCartFormUrl(content);
-        return content;
+        return fillBean(new ProductDetailPageContent(), product, variant);
+    }
+
+    protected <T extends ProductDetailPageContent> T fillBean(final T bean, final ProductProjection product, final ProductVariant variant) {
+        fillAdditionalTitle(product, bean);
+        fillProduct(product, variant, bean);
+        fillBreadCrumb(product, variant, bean);
+        fillAddToCartFormUrl(bean);
```

Comment:
`fillField` is a void method and the `initialize` method returns the bean and has type parameters